src/components/Card.js

```diff
@@ -1,45 +1,68 @@
-import React, { Component } from 'react';
-import LikeBadge from './LikeBadge'
-import SaleBadge from './SaleBadge'
+import React, { Component } from "react";
+import LikeBadge from "./LikeBadge";
+import SaleBadge from "./SaleBadge";
 
 class Card extends Component {
-    constructor(props) {
-        super(props)
-
-        this.state = {
-            likes: 0
-        }
-
-        this.onButtonClick = this.onButtonClick.bind(this)
-    }
-
-    onButtonClick(event) {
-        const likes = this.state.likes + 1
-        this.setState({likes})
-    }
-
-    render() {
-        const imageURL = `assets/images/${this.props.item.filenameString}`
-
-        return (
-            <div className="card">
-                <SaleBadge onSale={this.props.item.onSale}/>
-                <div className="card-img">
-                    <img src={imageURL} alt=""/>
-                </div>
-                <div className="card-details">
-                    <p className="card-item-tag">{this.props.item.tag}</p>
-                    <p className="card-item-name">{this.props.item.title}</p>
-                    <button className="card-item-like-btn" onClick={event => this.onButtonClick(event)}>
-                        <LikeBadge 
-                            likes={this.state.likes}
-                        />
-
-                    </button>
-                </div>
-            </div>
-        )
-    }
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      likes: 0
+    };
+
+    // This is one way to bind the event handler to this context
+    // We don't really use it at work, but I see it a lot in various packages
+    this.onButtonClick = this.onButtonClick.bind(this);
+  }
+
+  // Another way to define this would be using a babel class property, which
+  // automatically binds it to this if you use an arrow function, which means you don't need
+  // the line in the constructor. This is mostly what we use, but up to team preference
+
+  // onButtonClick = event => {
+  onButtonClick(event) {
```

Comment:
I love it. Having to bind everything in the `constructor()` is *annoying*. Thanks!